File: ios/SportsApp/DeviceIdModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(DeviceIdModule, NSObject)

RCT_EXTERN_REMAP_MODULE(DeviceIdModule, DeviceIdModule)

RCT_EXTERN_METHOD(getDeviceId:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/SportsApp/DeviceIdModule.swift
import Foundation

@objc(DeviceIdModule)
class DeviceIdModule: NSObject {

  @objc
  func getDeviceId(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    if let uuid = UIDevice.current.identifierForVendor?.uuidString {
      resolve(uuid)
    } else {
      reject("DEVICE_ID_ERROR", "Unable to get identifierForVendor", nil)
    }
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
